src/steps/do-query/do-get.js

```diff
@@ -63,20 +64,56 @@ export default function(requestContext, responseContext, registry) {
 
 }
 
-function parseFields(fieldsParam) {
+function parseFields(fieldsParam, registry) {
   let fields;
   if(typeof fieldsParam === "object") {
     fields = {};
     let isField = (it) => !arrayContains(["id", "type"], it);
 
     for(let type in fieldsParam) {
-      let provided = parseCommaSeparatedParam(fieldsParam[type]) || [];
+      let provided = parseCommaSeparatedParam(fieldsParam[type], type, registry) || [];
       fields[type] = provided.filter(isField);
     }
   }
   return fields;
 }
 
-function parseCommaSeparatedParam(it) {
-  return it ? it.split(",").map(decodeURIComponent) : undefined;
+function parseCommaSeparatedParam(it, type, registry) {
+  return it ? it.split(",").map(f => {
+    return transformDotSeparatedPath(f, type, registry);
+  }) : undefined;
+}
+
+function transformDotSeparatedPath(it, type, registry) {
```

Comment:
There's another complication here. Imagine that someone does `GET /organizations?sort=name` and, as in the example repo, that endpoint returns a bunch of different organization subtypes (schools, hotels, whatever). Now, imagine that some of those subtypes have different rules for dasherization than others. What happens? Well, the `type` argument this function will receive will be `organizations`, so it'll look up the dasherization rules for name with `shouldCamelize("organizations", registry)`...even though some of the resources won't be (just) organizations but schools or whatever with different dasherization rules. 

I need to think more about how to handle this (at the adapter level too, since it wasn't really designed with these cases in mind). We may need to lay down some constraints to simplify the problem, or add some more powerful abstractions. I'm not sure yet.